src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/CaseSearchRequest.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.ccd.domain.service.search.elasticsearch;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.ccd.domain.service.common.ObjectMapperService;
+import uk.gov.hmcts.ccd.endpoint.exceptions.BadSearchRequest;
+
+/**
+ * Sample ES json search request.
+ * <p>
+ * {
+ * 	"query": {
+ * 		"bool": {
+ * 			"filter": {
+ * 				"match": { "state": "AwaitingPayment"}
+ *                }
+ *            }
+ *       },
+ * 	"sort": {
+ * 		"id": { "order":"asc" }
+ *       }
+ * }
+ */
+@Slf4j
+public class CaseSearchRequest {
+
+    private static final String QUERY_NAME = "query";
+
+    private final ObjectMapperService objectMapperService;
+    private final String caseTypeId;
+    private final ObjectNode searchRequestJsonNode;
+
+
+    public CaseSearchRequest(ObjectMapperService objectMapperService, String caseTypeId, String jsonSearchRequest) {
```

Comment:
1) Can we make this class immutable? i.e. no setter. It'll contain the original request only.
When we create the secured one, we'll create a new instance of CaseSearchRequest which contains the secured query

Reply:
2. Shall we leave the objectMapperService outside of this class? i.e. the client has to build a ObjectNode and pass it in.

Reply:
changes done. CaseSearchRequest is now immutable